common/src/tap_manager.rs

```diff
@@ -106,6 +143,84 @@ impl TapManager {
 
         Ok(())
     }
+
+    async fn sender_denylist_reload(
+        pgpool: PgPool,
+        denylist_rwlock: Arc<RwLock<HashSet<Address>>>,
+    ) -> anyhow::Result<()> {
+        // Fetch the denylist from the DB
+        let sender_denylist = sqlx::query!(
+            r#"
+                SELECT sender_address FROM scalar_tap_denylist
+            "#
+        )
+        .fetch_all(&pgpool)
+        .await?
+        .iter()
+        .map(|row| Address::from_str(&row.sender_address))
+        .collect::<Result<HashSet<_>, _>>()?;
+
+        *(denylist_rwlock.write().await) = sender_denylist;
+
+        Ok(())
+    }
+
+    async fn sender_denylist_watcher(
+        pgpool: PgPool,
+        mut pglistener: PgListener,
+        denylist: Arc<RwLock<HashSet<Address>>>,
+    ) {
+        #[derive(serde::Deserialize)]
+        struct DenylistNotification {
+            tg_op: String,
+            sender_address: Address,
+        }
+
+        loop {
+            let pg_notification = pglistener.recv().await.expect(
+                "should be able to receive Postgres Notify events on the channel \
+                'scalar_tap_deny_notification'",
+            );
+
```

Comment:
Ideally, you would have a select! {} to handle cancelations. Your task should have a CancellationToken that you can `break` the loop so you can safely do a `handle.join()` on Drop.

Reply:
This might not be required for this task since we don't need to finish the computation before shutting the task down. But it can be considered a good practice to handle all JoinHandle.

Reply:
Yeah I don't think it's super critical here. Nevertheless I took a stab at making it a bit cleaner in 8863e0a1d7e5b3b8b0b29527f05c0b452bdbac7d